BLSQ/iaso
# IA-1983: registry

New registry domain.

[Mock-ups](https://whimsical.com/microplanning-cmr-9o1A7M8MaV6H5Bm2guUD4t) (only two first screens)

An org unit can be defined with a reference submission. You can basically add a submission to an org unit.
Org unit is normally of a specific org unit type.
This org unit type has sub org unit types.
The idea is to show: 
-  an org unit with his reference submission 
- all the submissions made of a specific form on the children org units  (included in the sub org unit types)

For the reference submission i used this form:
[microplan_obc_2022112302.xlsx](https://github.com/BLSQ/iaso/files/11087553/microplan_obc_2022112302.xlsx)

For the reference form of the children i used this form:
[microplan_localite_2022112301.xlsx](https://github.com/BLSQ/iaso/files/11087557/microplan_localite_2022112301.xlsx)

Related JIRA tickets : IA-1983

## Self proofreading checklist


## Changes

- new registry domain
- new `useEnketoUrl` hook
- new tools to calculate map Bounds in `hat/assets/js/apps/Iaso/utils/mapUtils.ts`

## How to test

- Create two new forms with the XLSX provided.
- Edit the second form in the admin and add a few `label_keys`
- make a submission for the first one and a few of the second
(you can by pass those creation and use existing submissions.)
- Select an org unit having a shape, children with geolocation
- Make sure the org unit type of this org unit include the sub org unit types of the previous children
- attach the main submission to the main org unit with the admin (edit the org unit and provide the instance id)
- attach the children submissions to the children org unit too
- Navigate to the Registry entry in the dashboard menu
![Screenshot 2023-03-28 at 11 11 27](https://user-images.githubusercontent.com/12494624/228191043-672237eb-cc46-4c55-bb3f-ff45a1d0abef.png)

- Select the main org unit (here "Aketi Zone de Santé") in the tree and click on See registry 
![Screenshot 2023-03-28 at 11 21 18](https://user-images.githubusercontent.com/12494624/228191102-8a253253-63aa-43e7-a7e8-8203f68a3493.png)

- Select the form used to create children submission in the form selector
![Screenshot 2023-03-28 at 11 22 39](https://user-images.githubusercontent.com/12494624/228195119-4975042a-0077-4775-9565-6528346e8644.png)

- You should be able to see instances list with pre selected columns (the `label_keys`)
- Play with the map, the table, the visible columns and with enketo if enabled.

PS: I have translations issue with the second form, there is a [JIRA](https://bluesquare.atlassian.net/browse/IA-2012) ticket for it
